### PR TITLE
Handle workspaces empty state

### LINKED ISSUE #841: Workspaces empty state
If on reload or on sync request user does not have access to any workspace 

- [ ] Bring up "Create a workspace" message (Design based on login form) 
- [ ] **Copy:**
   Body:
   >You no longer have access to your last workspace. Create a new one to continue using Toggl

   Field helper text:
   >Workspace name

   Button:
   > Create workspace
 
screen shot 2017-06-19 at 12 16 31

- [ ] Keep buttons Toggl.com, Log out, Settings and Sync active
- [ ] When clicking "Create Workspace" button, create a new Workspace 
- [ ] If no Workspace name was entered, show field validation error: "Enter a workspace name"

### PR DESCRIPTION
**Pull request by shantanuraj**
Closes #841 

* Add a view for creating a new workspace
* Show validation error if no workspace name is entered
* Show create workspace prompt if a workspace is removed on sync
* Show create workspace prompt if a timer from a revoked workspace is resumed

## How to test

1. Starting with two toggl accounts (a, b)
2. Create a workspace with one account (a) and invite the other account (b)
3. Open the popup once after being added to the workspace (and accepting the invitation) so that sync is called an the button knows of the new workspace
4. Now deactivate or delete account b from a's workspace
5. Open the popup again this time you should be prompted to create a workspace

### CODE REVIEW

**Comment by shantanuraj** [comment]
For the lint failing, we need to update our lint rules to support ES6 and more, I'll make make a different issue/pr for that.

Have improved the ws check it should work now 👍

**Comment by LulzAugusto** [comment]
screen shot 2018-06-01 at 2 04 15 pm

Looks like it just kept the same size it was before displaying the form, but I still dunno why it worked for the second time. If you think that's just a weird glitch, feel free to dismiss the review and merge. 👍

**Comment by shantanuraj** [comment]
Thanks, I'll try my best to reproduce; in any case will still add a min-height so this doesn't happen. 👍

**Comment by shantanuraj** [comment]
@LulzAugusto Couldn't reproduce it exactly but added min-height; which guarantees that it will display correctly, regardless of the rest of previous view's size.

File: src/scripts/background.js [hunk 1]
```diff
@@ -1660,6 +1674,34 @@ var TogglButton = {
...
+  },
+
+  hasWorkspaceBeenRevoked: function (workspaces) {
+    if (!TogglButton.$user) {
+      return false;
+    }
+    const oldWorkspaces = TogglButton.$user.workspaces.map(ws => ws.id);
+    const newWorkspaces = (workspaces || []).map(ws => ws.id);
+    const isNotIn = arr => e => !arr.includes(e);
+    const isRevoked = oldWorkspaces.some(isNotIn(newWorkspaces));
```
[the remark this answers is not in the document]
Comment (shantanuraj) [reply src/scripts/background.js]: Sorry my brain parsed “any workspace” from #841 weirdly :D
Fixing it 🙏

[the remark this answers is not in the document]
Comment (shantanuraj) [reply src/scripts/background.js]: Fixed here bf7ab1116672d9a25a3ac8774007405ef45b5577

**Pull request merged**